File: orig/man/wordcount.m
{ Copyright (C) 1981 by Bell Laboratories, Inc., and Whitesmiths Ltd. }
.NM wordcount count words in input
.SY
.UL wordcount
.FU
.UL wordcount
counts the words in its input and writes the total as a line of
text to the output.
A ``word'' is a maximal sequence of characters not
containing a blank or tab or newline.
.EG
.Q1
wordcount
A single line of input.
<ENDFILE>
.S 5
.Q2
.BU
The definition of ``word'' is simplistic.

File: orig/intro/wordcount.cc
#include "../p2c.h"
#include "intro.h"

// wordcount -- count words in standard input
void wordcount()
{
    integer nw;
    character c;
    boolean inword;

    nw = 0;
    inword = false;
    while (getc(c) != ENDFILE)
        if ((c == BLANK) or (c == NEWLINE) or (c == TAB))
            inword = false;
        else if (not inword) {
            inword = true;
            nw = nw + 1;
        }
    putdec(nw, 1);
    putc(NEWLINE);
}
